src/frontend/src/session/cursor_manager.rs

```diff
@@ -452,6 +486,7 @@ impl SubscriptionCursor {
                 }
             }
         }
+        self.last_fetch = Instant::now();
```

Comment:
I think we will miss when `next` returns early in L449 and L449

Reply:
Since this cursor should be closed after 449 is returned, set last_fetch doesn't seem to make sense

Reply:
> Since this cursor should be closed after 449 is returned, set last_fetch doesn't seem to make sense

I think the cursor will not be closed but will be set to invalid state and remain in the cursor map. Actually after walking through the codes again, I found out that if [`initiate_query`](https://github.com/risingwavelabs/risingwave/blob/3e7d5c2ea2bc14158f6f8c8dfff45fcc530fb2d0/src/frontend/src/session/cursor_manager.rs#L308) or [`try_refill_remaining_rows`](https://github.com/risingwavelabs/risingwave/blob/3e7d5c2ea2bc14158f6f8c8dfff45fcc530fb2d0/src/frontend/src/session/cursor_manager.rs#L325) return an error, cursor state will not be set to invalid so I am thinking whether we should set the state to invalid in `next` instead of `next_row` to make sure the all errors are fully covered.

Reply:
Also, I think we can further improve the error handling via one of the following ideas:
1. Auto close the cursor if the cursor is in an invalid state.
2. Auto retry/recreate the query stream / RPC when meeting errors

Reply:
will do it in next pr